# Análise Exploratória dos Dados

In [101]:
import pandas as pd
from pathlib import Path
import seaborn as sns
import matplotlib.pyplot as plt 
import numpy as np

In [102]:
caminho_arquivo = Path(r'C:\Users\alexa\OneDrive\Área de Trabalho\projeto_a3\dados_tratados\obesity_cleaned2.csv')

In [103]:
dados = pd.read_csv(caminho_arquivo)
del dados['Unnamed: 0']
dados


Country  Year     Obesity (%)         Sex Region
0      Afghanistan  2016   5.5 [3.4-8.1]  Both Sexes  Other
1      Afghanistan  2016   3.2 [1.3-6.4]        Male  Other
2      Afghanistan  2016  7.6 [4.3-12.4]      Female  Other
3      Afghanistan  2015   5.2 [3.3-7.7]  Both Sexes  Other
4      Afghanistan  2015   3.0 [1.3-6.0]        Male  Other
...            ...   ...             ...         ...    ...
24565     Zimbabwe  1976   0.5 [0.1-1.3]        Male  Other
24566     Zimbabwe  1976  7.0 [3.3-12.4]      Female  Other
24567     Zimbabwe  1975   3.7 [1.7-6.5]  Both Sexes  Other
24568     Zimbabwe  1975   0.4 [0.1-1.3]        Male  Other
24569     Zimbabwe  1975  6.7 [3.0-12.2]      Female  Other

[24570 rows x 5 columns]

### Os dados entre homens e mulheres são parecidos? Informe com números.

In [104]:
dados.dtypes

Country        object
Year            int64
Obesity (%)    object
Sex            object
Region         object
dtype: object

In [105]:
dados['Obesity_mean'] = dados['Obesity (%)'].str.split('[').str[0]

In [106]:
dados.dtypes

Country         object
Year             int64
Obesity (%)     object
Sex             object
Region          object
Obesity_mean    object
dtype: object

In [107]:
dados.Country

0        Afghanistan
1        Afghanistan
2        Afghanistan
3        Afghanistan
4        Afghanistan
            ...     
24565       Zimbabwe
24566       Zimbabwe
24567       Zimbabwe
24568       Zimbabwe
24569       Zimbabwe
Name: Country, Length: 24570, dtype: object

In [108]:
dados['Obesity_mean'] = pd.to_numeric(dados['Obesity_mean'], errors='coerce')
dados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24570 entries, 0 to 24569
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Country       24570 non-null  object 
 1   Year          24570 non-null  int64  
 2   Obesity (%)   24570 non-null  object 
 3   Sex           24570 non-null  object 
 4   Region        24570 non-null  object 
 5   Obesity_mean  24066 non-null  float64
dtypes: float64(1), int64(1), object(4)
memory usage: 1.1+ MB


In [109]:
dados.head()

Country  Year     Obesity (%)         Sex Region  Obesity_mean
0  Afghanistan  2016   5.5 [3.4-8.1]  Both Sexes  Other           5.5
1  Afghanistan  2016   3.2 [1.3-6.4]        Male  Other           3.2
2  Afghanistan  2016  7.6 [4.3-12.4]      Female  Other           7.6
3  Afghanistan  2015   5.2 [3.3-7.7]  Both Sexes  Other           5.2
4  Afghanistan  2015   3.0 [1.3-6.0]        Male  Other           3.0

In [110]:
media_homens = dados[dados['Sex'] == 'Male']
media_homens = media_homens['Obesity_mean'].mean()
media_homens

9.332984293193716

A média de obesidade mundial dos homens é de 9.33%

In [111]:
media_mulheres = dados[dados['Sex'] == 'Female']
media_mulheres = media_mulheres['Obesity_mean'].mean()
media_mulheres

15.533433059087509

A média de obesidade mundial das mulheres é de 15.53%

In [112]:
amplitude_homens_mulheres = media_mulheres - media_homens
amplitude_homens_mulheres

6.200448765893793

A diferença do percetual de gordura mundial entre homens e mulheres é de 6.20%, ou seja, as mulheres tem tendência de ter maior percentual de gordura do que os homens.

### Qual o percentual médio de obesidade por sexo na América do Norte no ano de 2010

In [113]:
america_do_norte_2010 = dados[(dados['Region'] == 'North America') & (dados['Year'] == 2010)]
america_do_norte_2010

Country  Year       Obesity (%)         Sex  \
3924                     Canada  2010  26.0 [23.2-28.9]  Both Sexes   
3925                     Canada  2010  25.8 [21.8-30.1]        Male   
3926                     Canada  2010  26.2 [22.4-30.2]      Female   
13878                    Mexico  2010  25.8 [23.1-28.6]  Both Sexes   
13879                    Mexico  2010  21.1 [17.5-25.0]        Male   
13880                    Mexico  2010  29.8 [25.8-33.9]      Female   
23454  United States of America  2010  32.3 [29.6-35.1]  Both Sexes   
23455  United States of America  2010  31.2 [27.4-35.3]        Male   
23456  United States of America  2010  33.3 [29.6-37.1]      Female   

              Region  Obesity_mean  
3924   North America          26.0  
3925   North America          25.8  
3926   North America          26.2  
13878  North America          25.8  
13879  North America          21.1  
13880  North America          29.8  
23454  North America          32.3  
23455  North America          31.2  
23456  North America          33.3

In [114]:
media_homens_na = america_do_norte_2010[america_do_norte_2010['Sex'] == 'Male']
media_homens_na.Obesity_mean.mean()

26.033333333333335

A média do percentual de pessoas do sexo masculino na América do norte no ano de 2010 é de 26.03%

In [115]:
media_mulheres_na = america_do_norte_2010[america_do_norte_2010['Sex'] == 'Female']
media_mulheres_na.Obesity_mean.mean()

29.766666666666666

A média do percentual de pessoas do sexo masculino na América do norte no ano de 2010 é de 29.77%

### Qual os top 3 com maior e menor taxa de aumento de índices de obesidade nesse período de 2010? E em 2016?

In [116]:
dados

Country  Year     Obesity (%)         Sex Region  Obesity_mean
0      Afghanistan  2016   5.5 [3.4-8.1]  Both Sexes  Other           5.5
1      Afghanistan  2016   3.2 [1.3-6.4]        Male  Other           3.2
2      Afghanistan  2016  7.6 [4.3-12.4]      Female  Other           7.6
3      Afghanistan  2015   5.2 [3.3-7.7]  Both Sexes  Other           5.2
4      Afghanistan  2015   3.0 [1.3-6.0]        Male  Other           3.0
...            ...   ...             ...         ...    ...           ...
24565     Zimbabwe  1976   0.5 [0.1-1.3]        Male  Other           0.5
24566     Zimbabwe  1976  7.0 [3.3-12.4]      Female  Other           7.0
24567     Zimbabwe  1975   3.7 [1.7-6.5]  Both Sexes  Other           3.7
24568     Zimbabwe  1975   0.4 [0.1-1.3]        Male  Other           0.4
24569     Zimbabwe  1975  6.7 [3.0-12.2]      Female  Other           6.7

[24570 rows x 6 columns]

In [117]:
paises_1975_dois_sexos = dados[(dados['Sex'] == 'Both Sexes') & (dados['Year'] == 1975)]
paises_1975_dois_sexos


Country  Year      Obesity (%)         Sex  \
123                           Afghanistan  1975    0.5 [0.2-1.1]  Both Sexes   
249                               Albania  1975   6.5 [3.6-10.5]  Both Sexes   
375                               Algeria  1975   6.9 [3.9-10.9]  Both Sexes   
501                               Andorra  1975  12.9 [8.6-18.1]  Both Sexes   
627                                Angola  1975    0.8 [0.3-1.9]  Both Sexes   
...                                   ...   ...              ...         ...   
24063  Venezuela (Bolivarian Republic of)  1975   9.6 [5.9-14.1]  Both Sexes   
24189                            Viet Nam  1975    0.1 [0.0-0.3]  Both Sexes   
24315                               Yemen  1975    2.8 [1.3-5.4]  Both Sexes   
24441                              Zambia  1975    1.5 [0.7-3.0]  Both Sexes   
24567                            Zimbabwe  1975    3.7 [1.7-6.5]  Both Sexes   

      Region  Obesity_mean  
123    Other           0.5  
249    Other           6.5  
375    Other           6.9  
501    Other          12.9  
627    Other           0.8  
...      ...           ...  
24063  Other           9.6  
24189  Other           0.1  
24315  Other           2.8  
24441  Other           1.5  
24567  Other           3.7  

[195 rows x 6 columns]

In [118]:
paises_2010_dois_sexos = dados[(dados['Sex'] == 'Both Sexes') & (dados['Year'] == 2010)]
paises_2010_dois_sexos

Country  Year       Obesity (%)         Sex  \
18                            Afghanistan  2010     4.0 [2.5-5.9]  Both Sexes   
144                               Albania  2010  18.2 [14.3-22.3]  Both Sexes   
270                               Algeria  2010  23.3 [19.3-27.6]  Both Sexes   
396                               Andorra  2010  24.4 [19.7-29.5]  Both Sexes   
522                                Angola  2010     6.2 [3.8-9.2]  Both Sexes   
...                                   ...   ...               ...         ...   
23958  Venezuela (Bolivarian Republic of)  2010  23.3 [19.7-27.1]  Both Sexes   
24084                            Viet Nam  2010     1.4 [0.9-1.9]  Both Sexes   
24210                               Yemen  2010  13.5 [10.6-16.7]  Both Sexes   
24336                              Zambia  2010     6.3 [4.6-8.2]  Both Sexes   
24462                            Zimbabwe  2010  13.8 [11.0-16.8]  Both Sexes   

      Region  Obesity_mean  
18     Other           4.0  
144    Other          18.2  
270    Other          23.3  
396    Other          24.4  
522    Other           6.2  
...      ...           ...  
23958  Other          23.3  
24084  Other           1.4  
24210  Other          13.5  
24336  Other           6.3  
24462  Other          13.8  

[195 rows x 6 columns]

In [119]:
df_taxa = paises_2010_dois_sexos[['Country', 'Obesity_mean']]
df_taxa.rename(columns={"Obesity_mean":"Obesity_2010"}, inplace=True)
df_taxa

C:\Users\alexa\AppData\Local\Temp\ipykernel_5004\2942407590.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_taxa.rename(columns={"Obesity_mean":"Obesity_2010"}, inplace=True)


Country  Obesity_2010
18                            Afghanistan           4.0
144                               Albania          18.2
270                               Algeria          23.3
396                               Andorra          24.4
522                                Angola           6.2
...                                   ...           ...
23958  Venezuela (Bolivarian Republic of)          23.3
24084                            Viet Nam           1.4
24210                               Yemen          13.5
24336                              Zambia           6.3
24462                            Zimbabwe          13.8

[195 rows x 2 columns]

In [123]:
df_taxa['Obesity_1975'] = paises_1975_dois_sexos[['Obesity_mean']]
df_taxa


C:\Users\alexa\AppData\Local\Temp\ipykernel_5004\239154478.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_taxa['Obesity_1975'] = paises_1975_dois_sexos[['Obesity_mean']]


Country  Obesity_2010  Obesity_1975
18                            Afghanistan           4.0           NaN
144                               Albania          18.2           NaN
270                               Algeria          23.3           NaN
396                               Andorra          24.4           NaN
522                                Angola           6.2           NaN
...                                   ...           ...           ...
23958  Venezuela (Bolivarian Republic of)          23.3           NaN
24084                            Viet Nam           1.4           NaN
24210                               Yemen          13.5           NaN
24336                              Zambia           6.3           NaN
24462                            Zimbabwe          13.8           NaN

[195 rows x 3 columns]

In [ ]:
df_taxa

Country  Obesity_2010  Obesity_1975
18                            Afghanistan           4.0           NaN
144                               Albania          18.2           NaN
270                               Algeria          23.3           NaN
396                               Andorra          24.4           NaN
522                                Angola           6.2           NaN
...                                   ...           ...           ...
23958  Venezuela (Bolivarian Republic of)          23.3           NaN
24084                            Viet Nam           1.4           NaN
24210                               Yemen          13.5           NaN
24336                              Zambia           6.3           NaN
24462                            Zimbabwe          13.8           NaN

[195 rows x 3 columns]

In [126]:
lista_obesidade_1975 = list(paises_1975_dois_sexos['Obesity_mean'])


In [128]:
df_taxa['Obesity_1975'] = lista_obesidade_1975

C:\Users\alexa\AppData\Local\Temp\ipykernel_5004\3569600128.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_taxa['Obesity_1975'] = lista_obesidade_1975


In [138]:
df_taxa['Taxa_aumento_%'] = ((df_taxa['Obesity_2010'] - df_taxa['Obesity_1975']) / df_taxa['Obesity_1975']) * 100
df_taxa

C:\Users\alexa\AppData\Local\Temp\ipykernel_5004\2342464556.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_taxa['Taxa_aumento_%'] = ((df_taxa['Obesity_2010'] - df_taxa['Obesity_1975']) / df_taxa['Obesity_1975']) * 100


Country  Obesity_2010  Obesity_1975  \
18                            Afghanistan           4.0           0.5   
144                               Albania          18.2           6.5   
270                               Algeria          23.3           6.9   
396                               Andorra          24.4          12.9   
522                                Angola           6.2           0.8   
...                                   ...           ...           ...   
23958  Venezuela (Bolivarian Republic of)          23.3           9.6   
24084                            Viet Nam           1.4           0.1   
24210                               Yemen          13.5           2.8   
24336                              Zambia           6.3           1.5   
24462                            Zimbabwe          13.8           3.7   

       Taxa_aumento_%  
18         700.000000  
144        180.000000  
270        237.681159  
396         89.147287  
522        675.000000  
...               ...  
23958      142.708333  
24084     1300.000000  
24210      382.142857  
24336      320.000000  
24462      272.972973  

[195 rows x 4 columns]

In [152]:
df_taxa.sort_values('Taxa_aumento_%', ascending=True, inplace=True)

C:\Users\alexa\AppData\Local\Temp\ipykernel_5004\3853848381.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_taxa.sort_values('Taxa_aumento_%', ascending=True, inplace=True)


Os países Top 3 que tiveram a maior taxa de crescimento do percentual de obesidade, considerando de 1975 até foi:

In [153]:
df_taxa.iloc[:3]

Country  Obesity_2010  Obesity_1975  Taxa_aumento_%
15012               Nauru          58.9          44.5       32.359551
18036  Russian Federation          21.4          13.9       53.956835
7200              Estonia          19.6          12.7       54.330709

In [156]:
df_taxa.sort_values('Taxa_aumento_%', ascending=True, inplace=True)
df_taxa

C:\Users\alexa\AppData\Local\Temp\ipykernel_5004\2509507454.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_taxa.sort_values('Taxa_aumento_%', ascending=True, inplace=True)


Country  Obesity_2010  Obesity_1975  Taxa_aumento_%
15012               Nauru          58.9          44.5       32.359551
18036  Russian Federation          21.4          13.9       53.956835
7200              Estonia          19.6          12.7       54.330709
11862              Latvia          21.8          13.4       62.686567
13374               Malta          26.8          15.6       71.794872
...                   ...           ...           ...             ...
24084            Viet Nam           1.4           0.1     1300.000000
14130              Monaco           NaN           NaN             NaN
18792          San Marino           NaN           NaN             NaN
20430         South Sudan           NaN           NaN             NaN
20808               Sudan           NaN           NaN             NaN

[195 rows x 4 columns]

Os países Top 3 que tiveram a menor taxa de crescimento do percentual de obesidade, considerando de 1975 até 2010 foi:

In [157]:
df_taxa.iloc[:3]

Country  Obesity_2010  Obesity_1975  Taxa_aumento_%
15012               Nauru          58.9          44.5       32.359551
18036  Russian Federation          21.4          13.9       53.956835
7200              Estonia          19.6          12.7       54.330709